### PR TITLE
[Hot Fix 🔧 ] Fix Duplicating on submit Remove Patient with Reason

### LINKED ISSUE #3142: Remove patient overflow menu and editing registration info item duplicates patient in household profile
**Describe the bug**

**To Reproduce**
**Scenario 1**
Steps to reproduce the behavior: Remove patient
1. Log in to any app  whose workflow begins with household register
2. Open any household with patients
3. Click on any patient and open overflow menu items
4. Click remove patient and fill in the questionnaire and save it
5. Go to household profile . You will notice that the patient you remove has duplicated on the household profile

**Scenario 2**
Steps to reproduce the behavior: Edit registration info
1. Log in to any app  whose workflow begins with household register
2. Open any household with patients
3. Click on any patient and open overflow menu items
4. Click registration info and edit any field in the questionnaire and save it
5. Go to household profile . You will notice that the patient you edit has duplicated on the household profile

**Expected behavior**
Removing a patient from household should grey out the patient or remove him/her from the household profile based on different app but not duplicate the patient
Editing patient details should save the changes but not duplicate the patient

**Screenshots**.

image

**Smartphone (please complete the following information):**
 - Device: [e.g. Samsung Galaxy A21 S]
 - Android version: 11
 - Version 30

**Additional context**

### PR DESCRIPTION
**Pull request by f-odhiambo**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #3142 

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

**Comment by f-odhiambo** [comment]
No test written for this PR just initial commit 
CC @pld @SebaMutuku @allan-on @ndegwamartin

**Comment by SebaMutuku** [comment]
@f-odhiambo I think we tested this and works. I'm okay with the solution.

**Review by pld — APPROVED** [review]

File: android/engine/src/main/java/org/smartregister/fhircore/engine/data/local/DefaultRepository.kt [hunk 1]
```diff
@@ -671,9 +671,17 @@ constructor(
...
+              ?.flatten()
+              ?.distinctBy { it.id }
+              ?.groupBy { it.resourceType }
           val reverseIncludedResources: Map<ResourceType, List<Resource>>? =
-            currentSearchResult.revIncluded?.values?.flatten()?.groupBy { it.resourceType }
+            currentSearchResult.revIncluded
+              ?.values
+              ?.flatten()
+              ?.distinctBy { it.id }
+              ?.groupBy { it.resourceType }
```
Comment (pld) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/data/local/DefaultRepository.kt:684]: please add a todo into the code to remove this after https://github.com/google/android-fhir/issues/2512 is merged

**Review by pld — CHANGES_REQUESTED** [review]

File: android/engine/src/main/java/org/smartregister/fhircore/engine/data/local/DefaultRepository.kt (hunk 1 above)
Comment (f-odhiambo) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/data/local/DefaultRepository.kt:684]: Done

**Review by pld — APPROVED** [review]

**Review by pld — APPROVED** [review]

**Pull request merged**